### PR TITLE
Fix desktop app stuck in skeleton loading state

### PR DESCRIPTION
**Pull request by 2witstudios**
The desktop app was getting stuck showing an infinite skeleton loader when
loading pages. Root cause: auth endpoints used by desktop (mobile login,
device refresh, desktop OAuth exchange) were not setting session cookies.
The Next.js middleware requires a session cookie for page route requests,
so without one, page loads would fail silently and the SWR fetch would
hang without surfacing an error.

### Core fix: Session cookies for desktop auth endpoints
- Set session cookie in mobile login endpoint when platform is 'desktop'
- Set session cookie in device refresh endpoint when platform is 'desktop'
- Set session cookie in desktop OAuth exchange endpoint
- Set session cookie in Google OAuth exchange when platform is 'desktop'
- Add credentials: 'include' to desktop login fetch for cookie handling

### Cookie-expiry deadlock fix
- Add device/mobile/desktop auth routes to middleware public routes list
  so device refresh can recover from expired cookies (these endpoints
  authenticate via body tokens, not session cookies)
- Propagate OAuth exchange cookie to BrowserWindow session via
  `session.defaultSession.cookies.set()` (Node.js main-process fetch
  doesn't share cookies with the renderer)

### Loading UX improvements
- Add 15s fetch timeout and SWR error retry config (3 retries, 2s delay)
- Add error state UI with retry button in CenterPanel
- Add loading timeout (12s) with "Taking longer than expected" hint
- Fix `loadingTimedOut` not resetting when Retry is clicked (`handleRetry` wrapper)
- Show loading skeleton during SWR retry windows (prevents "Page not found"
  during transient failures)
- Use shadcn `Button` component instead of raw `<button>` elements

### Desktop startup improvements
- Disable backgroundThrottling during startup, re-enable after ready-to-show
- Restore desktop active tab route on startup (useTabSync)
- Warm auth-fetch session cache from loadSession to eliminate redundant IPC
- Logout now fully resets tab state including pinned tabs

### Service worker hardening
- Skip `_next/` in service worker to prevent stale bundle mismatches
- Add 503 fallback for RSC fetch failures (offline/timeout resilience)
- Robust Electron service worker cleanup

### Code quality (CodeRabbit review feedback)
- Re-read store state after healing/restore in useTabSync
- Remove unnecessary `typeof window` check in useEffect
- Add null guard for `sessionClaims` on web device refresh path
- Add `isValidating` to SWR mock and `retry()` test coverage
- Add desktop restore edge case tests (one-shot guard, deep-link skip)
- Add clarifying comment for `retry` vs `invalidateTree` distinction

## How to validate
1. Desktop app loads without stuck skeleton
2. OAuth login on desktop sets cookie and navigates to dashboard
3. After >7 days idle, device refresh recovers without signin redirect
4. Transient network failures show loading skeleton during retries, not "Page not found"
5. All 506 unit tests pass, CI green

https://claude.ai/code/session_011dazNWCw89QmzeT8QGejAB

## Summary by CodeRabbit

* **New Features**
  * Faster desktop startup by preventing background throttling during initial render.
  * Desktop auth flows now set session cookies for smoother sign-in and middleware compatibility.
  * Loading timeout UI with "Taking longer than expected…" and Retry.
  * One-time restoration of the previously active tab on desktop startup.
  * Service worker updated for improved runtime asset handling and offline/online strategies.

* **Bug Fixes**
  * Prevented serving stale runtime assets from the service worker.

### CODE REVIEW

File: apps/desktop/src/main/index.ts [hunk 1]
```diff
@@ -1006,6 +1010,29 @@ async function handleAuthExchange(url: string): Promise<boolean> {
...
+      await session.defaultSession.cookies.set({
+        url: appUrl.origin,
+        name: 'session',
+        value: tokens.sessionToken,
+        path: '/',
+        httpOnly: true,
+        secure: !appUrl.origin.includes('localhost'),
+        sameSite: 'strict' as const,
+        expirationDate: Math.floor(Date.now() / 1000) + 7 * 24 * 60 * 60, // 7 days
+      });
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply apps/desktop/src/main/index.ts]: Fixed in 8d9a50a1. Updated the secure flag check to `\!appUrl.origin.includes('localhost') && \!appUrl.origin.includes('127.0.0.1')`, matching `getAppUrl()` behavior (lines 59, 75).
